drivers/ethos_u/src/ethosu_callback.c

```diff
@@ -7,6 +7,82 @@
 #include <soc_memory_map.h>
 #include <cmsis_core.h>
 
+#ifndef BOARD_OSPI_RAM_BASE
+#define BOARD_OSPI_RAM_BASE 0xA0000000
+#define BOARD_OSPI_RAM_SIZE 0x20000000
+#endif
+#ifndef BOARD_OSPI_FLASH_BASE
+#define BOARD_OSPI_FLASH_BASE 0xC0000000
+#define BOARD_OSPI_FLASH_SIZE 0x20000000
+#endif
+
+typedef struct address_range {
+	uintptr_t base;
+	uintptr_t limit;
+} address_range_t;
+
+static const address_range_t no_need_to_invalidate_areas[] = {
+	/* TCM is never cached */
+	{
+		.base = 0x00000000,
+		.limit = 0x01FFFFFF,
+	},
+	{
+		.base = 0x20000000,
+		.limit = 0x21FFFFFF,
+	},
+	/* MRAM should never change while running */
+	{
+		.base = CONFIG_FLASH_BASE_ADDRESS,
+		.limit = CONFIG_FLASH_BASE_ADDRESS + (CONFIG_FLASH_SIZE * 1024) - 1,
+	}};
+
+static bool check_need_to_invalidate(const void *p, size_t bytes)
+{
+	uintptr_t base = (uintptr_t)p;
+	if (bytes == 0) {
+		return false;
+	}
+	uintptr_t limit = base + bytes - 1;
+	for (unsigned int i = 0;
+	     i < sizeof no_need_to_invalidate_areas / sizeof no_need_to_invalidate_areas[0]; i++) {
+		if (base >= no_need_to_invalidate_areas[i].base &&
+		    limit <= no_need_to_invalidate_areas[i].limit) {
+			return false;
+		}
+	}
+	return true;
+}
+
+bool ethosu_area_needs_invalidate_dcache(const void *p, size_t bytes)
```

Comment:
Both these "needs" functions are dependent on the application and how it's set up its MPU, so I suggest making them weak so apps can override, with these providing general defaults.